specifications/secure-launch-specification.rst

```diff
@@ -0,0 +1,719 @@
+data structures that must be present, as well as a user-defined data structure.
+Whereas on AMD, only the layout of an AMD Secure Loader Block is defined and
+thus, any configuration structures are SLB implementation specific.
+
+To provide a unified experience in passing configuration and other meta-data to
+a TrenchBoot Secure Launch entry point for a kernel, this specification
+provides the platform-agnostic Secure Launch Resource Table (SLRT) along with
+details on how it will be implemented for each platform supported.
+
+Terminology
+===========
+
+Requirements Language
+---------------------
+
+The key words **"MUST"**, **"MUST NOT"**, **"REQUIRED"**, **"SHALL"**,
+**"SHALL NOT"**, **"SHOULD"**, **"SHOULD NOT"**, **"RECOMMENDED"**,  **"MAY"**,
+and **"OPTIONAL"** in this document are to be interpreted as described in
+:rfc:`2119`.
+
+
+Acronyms
+--------
+
+:DCE: Dynamic Configuration Environment (*eg. Intel ACM/AMD SLB*)
+:DLE: Dynamic Launch Event (*eg. Intel GETSEC[SENTER]/AMD SKINIT*)
+:DLME: Dynamic Launch Measured Environment (*eg. Operating System/Hypervisor*)
+:DRTM: Dynamic Root of Trust Measurement
+
+
+Secure Launch Architecture
+==========================
+
+
+Secure Launch aware Bootloader
+------------------------------
+
+A bootloader is Secure Launch aware if it understands how to load or provide a
+DLE Handler and configure an SLRT.
+
+Bootloader Types
+~~~~~~~~~~~~~~~~
+
+A Secure Launch aware bootloader can be categorized as one of three types,
+
+:monolithic: This is a bootloader that contains the DLE Handler.
+:initializing: This is a bootloader that loads an external DLE Handler and
+               executes a supplemental bootloader.
+:supplemental: This is a bootloader that is capable of calling a DLE Handler.
+
+Dynamic Launch Event Handler
+----------------------------
+
+The DLE (Dynamic Launch Event) Handler consumes the SLRT and invokes the
+platform's dynamic launch event. This specification seeks to standardize the
+invocation interface for the DLE Handler for each platform supported by
+TrenchBoot. This will provide a well-defined interface for DLE Handler
+implementors to follow and bootloaders to comply to ensure interoperability
+between implementations.
+
+Secure Launch Resource Table
+----------------------------
+
+The SLRT provides a platform-agnostic, standard format for providing
+information to the DLE Handler and for passing DRTM relevant information across
+the DLE to be available for use by the DCE and the DLME. The table **SHALL** be
+initialized by a bootloader and any subsequent bootloaders in the boot chain
+**MAY** append entries to the table. While the table is designed to be
+agnostic, the reality of DRTM hardware will drive a difference in the
+construction and setup of the table.
+
+Secure Launch Entry
+-------------------
+
+The Secure Launch Entry is a kernel entry point for the DLME that is platform
+aware. Its interface is dictated by the platform's Dynamic Launch
```

Comment:
What does it mean "platform aware"?

Reply:
It is an english idiom to mean that X is aware of and can handle the differences cause by Y. In this case it means that the implementation is written with awareness of the idiosyncrasies of the platform. Now one might want to say it is platform specific, but that could be interpreted to mean there is a unique entry for each platform, which is not the case. The main example is there will be one entry point for x86 which handles both Intel and AMD platforms.

Reply:
@dpsmith My point is not that I don't understand what "platform aware" words mean but because the text assumes the reader knows what is behind those words. I would be satisfied if you used your explanation to improve the section description for those who don't have a deep understanding of implementation.

Reply:
please review the rephrasing of this sentence.